laser_encoders/download_models.py

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# LASER  Language-Agnostic SEntence Representations
+# is a toolkit to calculate multilingual sentence embeddings
+# and to use them for document classification, bitext filtering
+# and mining
+#
+# -------------------------------------------------------
+#
+# This python script installs NLLB LASER2 and LASER3 sentence encoders from Amazon s3
+
+# default to download to current directory
+
+import argparse
+import logging
+import os
+import sys
+
+import requests
+from language_list import LANGUAGE_MAPPING, SPM_LANGUAGE
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s",
+)
+logger = logging.getLogger("preprocess")
+
+
+class LaserModelDownloader:
+    def __init__(self, model_dir: str):
+        self.model_dir = model_dir
+        self.url = "https://dl.fbaipublicfiles.com/nllb/laser/"
+
+    def download(self, url: str):
+        if os.path.exists(os.path.join(self.model_dir, os.path.basename(url))):
+            logger.info(f" - {os.path.basename(url)} already downloaded")
+        else:
+            logger.info(f" - Downloading {os.path.basename(url)}")
+            response = requests.get(url, stream=True)
+            with open(os.path.join(self.model_dir, os.path.basename(url)), "wb") as f:
+                for chunk in response.iter_content(chunk_size=1024):
+                    f.write(chunk)
+
+    def download_laser2(self):
+        self.download(f"{self.url}/laser2.pt")
+        self.download(f"{self.url}/laser2.spm")
+        self.download(f"{self.url}/laser2.cvocab")
+
+    def download_laser3(self, lang: str, version: str = "v1", spm: bool = False):
+        if lang not in LANGUAGE_MAPPING:
+            logger.info(
+                f"Unsupported language name: {lang}. Please specify a supported language name."
+            )
+            return
+
+        if len(LANGUAGE_MAPPING[lang]) > 1:
+            options = ", ".join(f"'{opt}'" for opt in LANGUAGE_MAPPING[lang])
+            logger.info(
+                f"Language '{lang}' has multiple options: {options}. Please specify."
+            )
+            return
+
+        lang = LANGUAGE_MAPPING[lang]
```

Comment:
Here we probably want to check if the language is in the LASER2 list, and if it is the case, download the LASER2 model for it instead. 
In other words: I want to support the case when the user just specifies the language, and we decide ourselves, whether to give a LASER2 or LASER3 model, whatever is available.

Reply:
There isn't a LASER2 list, please point me to it if I'm incorrect.

Reply:
The languages supported by LASER2 are in this list: https://github.com/facebookresearch/LASER/#supported-languages 
There are only their English names; a mapping between them and 3+4-letter codes can be found here: https://github.com/facebookresearch/flores/tree/main/flores200#languages-in-flores-200.

Reply:
Alright